### ISSUE TITLE
Cockpit should pre-populate some of the domain-join fields

### ISSUE THREAD

**Issue by sgallagher** [cockpit-project/cockpit#16390]
Page: Overview

When joining a domain, there are two fields that could be auto-filled by Cockpit under some circumstances.

I propose the following:

1. Domain Field - If the system hostname is a FQDN, Cockpit should call `realm discover <FQDN>` and if it discovers a matching domain, pre-populate the field (with the option to edit it, of course).
2. Domain Administrator Name - Once the FreeIPA or AD domain is identified, this field should be pre-populated with the default administrative user for that domain type. For FreeIPA, the default administrator is `admin`; for Active Directory it is `Administrator`.

**Comment by martinpitt** [comment]
Curious, we actually try to do exactly that. When I simply click on "Join realm", I get this with FreeIPA:
![image](https://user-images.githubusercontent.com/200109/134505268-0853e67e-db82-424c-9e86-7dbf039e359d.png)

i.e. it auto-detected the domain name and admin password from
```
# realm discover
cockpit.lan
  type: kerberos
  realm-name: COCKPIT.LAN
  domain-name: cockpit.lan
  configured: no
  server-software: ipa
  client-software: sssd
  required-package: freeipa-client
  required-package: oddjob
  required-package: oddjob-mkhomedir
  required-package: sssd
```
(well, of course it uses the D-Bus API, not the CLI, but same difference) and the green "contacted domain" shows that the "cockpit.lan" domain was indeed pinged successfully. This also works against (Samba) AD for me, where the user name is different:

![image](https://user-images.githubusercontent.com/200109/134505904-23e68a8a-50fa-4d83-8fe7-1a36ce35edd0.png)

```
# realm discover
cockpit.lan
  type: kerberos
  realm-name: COCKPIT.LAN
  domain-name: cockpit.lan
  configured: no
  server-software: active-directory
  client-software: sssd
  required-package: oddjob
  required-package: oddjob-mkhomedir
  required-package: sssd
  required-package: adcli
  required-package: samba-common-tools
```

However, one issue is that despite the pre-filled "e.g. admin" template, one still has to actually type "admin" into the user field. This is confusing, as this is *not* necessary for the domain field (not touching it at all usually works).

So as a first step, I'd like to confirm with you whether you see the same:
 - domain+user gets detected, shown in gray with "e.g." prefix, but its values are correct
 - you forgot to actually type the user name

If domain detection did not work, I'll need some more debugging from you, but I'll think about that if/when necessary.

In case that really was the problem:
 * I think we should just drop the 'e.g.' and treat this as an "empty input line" template, and just pre-fill the actual values
 * Or, if we keep it as a template, DTRT if the user does not actually type anything into the user field, just like the domain field works.

**Comment by sgallagher** [comment]
You're correct; if the machine's hostname is right, it does give me 'Contacted domain'. I was testing both FreeIPA and AD and when I was filing this bug, I didn't have the domain matching.

From a UX perspective, the "e.g.' in it is very confusing and contradictory. Please drop the "e.g." and the quotes around the domain name.

I think the right behavior here would be to have a lightly-greyed-out default value in the admin name field that the user can overwrite if needed, but that if they don't bother to do so, it uses the default value.

**Comment by martinpitt** [comment]
Agreed, I think we should keep the "e.g." and the gray template values only for the case where no domain gets detected. Thanks for confirming! I'll look into that.

**Comment by martinpitt** [comment]
Fixed in #16409
